src/app/api/integrations/auth/callback/route.ts

```diff
@@ -0,0 +1,42 @@
+import { NextRequest, NextResponse } from "next/server";
+
+export async function GET(req: NextRequest) {
+  const { searchParams } = new URL(req.url);
+  const code = searchParams.get("code");
+  const state = searchParams.get("state");
+
+  if (!code) {
+    return NextResponse.json({ error: "Missing code" }, { status: 400 });
+  }
+
+  const basicAuth = Buffer.from(
+    `${process.env.NOTION_CLIENT_ID}:${process.env.NOTION_CLIENT_SECRET}`
```

Comment:
update `.env.example`, `.github/workflows/deploy-prod.yml`, `.github/workflows/deploy-staging.yml` and `Dockerfile` with new variables, make separate credentials for staging and prod environments, update those environment secrets on Github.